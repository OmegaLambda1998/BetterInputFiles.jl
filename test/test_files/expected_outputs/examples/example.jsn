{
    "INTERPOLATION": {
        "B": 1,
        "A": 1,
        "C": "Example"
    },
    "KEY1": {
        "B": 2,
        "A": 1,
        "SUBDICT": {
            "Y": 4,
            "SUBSUBDICT": [
                {
                    "Z": 5
                },
                {
                    "Z": 6
                }
            ],
            "X": 3
        }
    },
    "DEFAULT": {
        "EXAMPLE": "Example"
    },
    "GLOBAL": {
        "BASE_PATH": "/home/patricka/.julia/dev/BetterInputFiles/test/test_files/input_files/examples/",
        "INPUT_PATH": "/home/patricka/.julia/dev/BetterInputFiles/test/test_files/input_files/examples",
        "LOG_FILE": "/home/patricka/.julia/dev/BetterInputFiles/test/test_files/input_files/examples/Output/log.txt",
        "LOGGING": true,
        "OUTPUT_PATH": "/home/patricka/.julia/dev/BetterInputFiles/test/test_files/input_files/examples/Output"
    },
    "METADATA": {
        "ORIGINAL": "/home/patricka/.julia/dev/BetterInputFiles/test/test_files/input_files/examples/example.jsn",
        "A": "B",
        "DATE": "2023-03-01"
    },
    "ENV_VAR": {
        "B": 2,
        "A": 1
    }
}
