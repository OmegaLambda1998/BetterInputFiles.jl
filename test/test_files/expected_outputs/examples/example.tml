# Test default
[ default ]
example = "Example"

# Test update_case
[ key1 ]
a = 1
b = 2

# Test include
[ key2 ]
a = 1
b = 2

# Test env vars
[ env_var ]
a = 1
b = 2 

# Test interpolations
[ interpolation ]
a = 1
b = 1 
c = "Example" 

