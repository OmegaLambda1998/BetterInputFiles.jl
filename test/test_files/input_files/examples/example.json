{
    "default": {
        "example": "Example"
    },
    "key1": {
        "a": 1,
        "b": 2,
        "subdict": {
            "x": 3,
            "y": 4,
            "subsubdict": {
                "z": 5
            }
        }
    },
    <include includes/includes.json>
    "env_var": {
        "a": <$A>,
        "b": <$B>
    },
    "interpolation": {
        "a": 1,
        "b": <%a>,
        "c": <%example>
    }
}
