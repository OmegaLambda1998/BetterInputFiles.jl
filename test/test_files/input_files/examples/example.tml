# Test default
[ default ]
example = "Example"

# Test update_case
[ key1 ]
a = 1
b = 2

# Test include
<include includes/includes.toml>

# Test env vars
[ env_var ]
a = <$A>
b = <$B>

# Test interpolations
[ interpolation ]
a = 1
b = <%a>
c = <%example>

