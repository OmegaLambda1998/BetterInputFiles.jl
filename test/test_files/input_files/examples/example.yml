# Test default
default:
  example: "Example"


# Test update_case
key1:
  a: 1
  b: 2
  subdict:
    x: 3
    y: 4
    subsubdict:
      z: 5

# Test include
<include includes/includes.yaml>

# Test env vars
env_var:
  a: <$A>
  b: <$B>

# Test interpolations
interpolation:
  a: 1
  b: <%a>
  c: <%example>
