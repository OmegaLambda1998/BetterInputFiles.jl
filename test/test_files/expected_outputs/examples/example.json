{
    "default": {
        "example": "Example"
    },
    "key1": {
        "a": 1,
        "b": 2
    },
    "key2": {
        "a": 1,
        "b": 2
    },
    "env_var": {
        "a": 1,
        "b": 2 
    },
    "interpolation": {
        "a": 1,
        "b": 1,
        "c": "Example"
    }
}
